torch_xla/csrc/runtime/runtime.cpp

```diff
@@ -55,8 +55,18 @@ const absl::StatusOr<ComputationClient * absl_nonnull>& GetComputationClient() {
   // Since we only allow a single initialization, as soon as this function is
   // called, we store the initialization result in this trivially destructible
   // reference.
-  static const auto& maybe_client =
-      *new absl::StatusOr<ComputationClient*>(InitializeComputationClient());
+  static absl::StatusOr<std::unique_ptr<ComputationClient>> init_result =
```

Comment:
This violates Google's C++ style guide: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

For singleton objects, we deliberately do *not* want their destructors to be called, as that can lead to race condition at program exit time. 

I'm not sure what this PR is trying to achieve. Could you clarify why you want to sure that the PjRt client dtor is called? Usually we don't destroy the singleton objects - we just let the OS reclaim the resources when the process terminates.

Reply:
@zhanyong-wan thanks for the feedback. Could you give an example of the race condition you mentioned and why it was not addressed until v2.8?

Reply:
The style guide I mentioned noted: "When destructors are trivial, their execution is not subject to ordering at all (they are effectively not "run"); otherwise we are exposed to the risk of accessing objects after the end of their lifetime. Therefore, we only allow objects with static storage duration if they are trivially destructible. Fundamental types (like pointers and int) are trivially destructible, as are arrays of trivially destructible types."

For example, at program exit time there could be long-running threads accessing global variables. If a global variable is destructed, such access is undefined behavior.

As to why it wasn't addressed until v2.8, I don't know the history, but my guess is that we just noticed the potential race and decided to fix it.

Reply:
@rajkthakur , `StatusOr<T>` is not trivially destructible, regardless of whether `T` is trivially destructible. `PjrtComputationClient` is not trivially destructible and not meant to be. I don't understand what you mean by "we think PjrtComputationClient could be trivially destructible".